openedx/core/djangoapps/content_libraries/tests/test_api.py

```diff
@@ -13,8 +13,88 @@
     UsageKey,
 )
 from opaque_keys.edx.locator import LibraryLocatorV2
+from organizations.models import Organization
 
 from .. import api
+from ..constants import ALL_RIGHTS_RESERVED, COMPLEX
+from .base import requires_blockstore
+from openedx.core.lib.blockstore_api.methods import create_collection
+from openedx.core.djangolib.blockstore_cache import BundleCache
+
+
+@requires_blockstore
+class ApiTest(TestCase):
+    """
+    Integration tests for the content libraries api functions
+    """
+
+    def test_bundle_links(self):
+        """
+        Test robustness of getting bundle links
+
+        If a linked bundle is deleted, retrieving the list of linked bundles should not throw an error.
+        The non-existant linked bundle should be ignored.
+        """
+        collection = create_collection('blockstore test collection 1')
+        org = Organization.objects.create(name='whatever', short_name='whatever')
+        content_library = api.create_library(
+            collection.uuid,
+            COMPLEX,
+            org,
+            'abcd1234',
+            'Test',
+            'example description',
+            True,
+            True,
+            ALL_RIGHTS_RESERVED,
+        )
+        api.publish_changes(content_library.key)
+
+        assert api.get_bundle_links(content_library.key) == []
+
+        linked_library_1 = api.create_library(
+            collection.uuid,
+            COMPLEX,
+            org,
+            'abcd1235',
+            'Test2',
+            'example description 2',
+            True,
+            True,
+            ALL_RIGHTS_RESERVED,
+        )
+        block1 = api.create_library_block(linked_library_1.key, 'html', 'testblock1linked')
+        api.publish_changes(linked_library_1.key)
+
+        linked_library_2 = api.create_library(
+            collection.uuid,
+            COMPLEX,
+            org,
+            'abcd1236',
+            'Test3',
+            'example description 3',
+            True,
+            True,
+            ALL_RIGHTS_RESERVED,
+        )
+        block2 = api.create_library_block(linked_library_2.key, 'html', 'testblock2linked')
+        api.publish_changes(linked_library_2.key)
+
+        api.create_bundle_link(content_library.key, 'one', linked_library_1.key)
+        api.create_bundle_link(content_library.key, 'two', linked_library_2.key)
+
+        assert len(api.get_bundle_links(content_library.key)) == 2
+        assert (
+            set(map(lambda x: x.bundle_uuid, api.get_bundle_links(content_library.key)))
+            == set([linked_library_1.bundle_uuid, linked_library_2.bundle_uuid])
+        )
+
+        api.delete_library(linked_library_1.key)
+
+        BundleCache(linked_library_1.bundle_uuid).clear()
+        BundleCache(linked_library_2.bundle_uuid).clear()
+        assert len(api.get_bundle_links(content_library.key)) == 1
```

Comment:
@swalladge so this test fails without the fix you added?

Reply:
I've moved the test, but yes have confirmed that the new test failed without the fix, and passes with the fix.

Reply:
Confirmed -- this new test fails without the change added.

@swalladge  Could you put your instructions for running these tests in the PR here, so they're easy to find for next time?

Reply:
done :)